### PR TITLE
Merge cv32e40s

### PR DESCRIPTION
**Pull request by silabs-krdosvik**
Merge
Formal compiles
Hello world passes
ci check: passes except pma_cfg_2 tests

### CODE REVIEW

File: bsp/corev_uvmt.h
Comment (silabs-robin) [inline bsp/corev_uvmt.h:1]: This is my change. Looks familiar and good.

File: docs/VerifPlans/Simulation/privileged_spec/CV32E40SX_PMA_VerifPlan.json
Comment (silabs-robin) [inline docs/VerifPlans/Simulation/privileged_spec/CV32E40SX_PMA_VerifPlan.json:1]: I think I recall most of these vplan updates. Good to have them in.

One scary thing though, is for example those `if (!pmp)` things. They could hide that certain tests are in fact not run on the X.

File: tests/cfg/pma_test_cfg_2.yaml
Comment (silabs-robin) [inline tests/cfg/pma_test_cfg_2.yaml:5]: This, as we know, will break some tests. But we have the fix on the S and are ready to move it over right after this merge.

**Review by silabs-robin — APPROVED** [review]
Looks good.
I've commented on some of my reasoning, for the record.

Will merge when `ci check: ongoing` is completed.
(I don't care if it is not at 100%, because we know of the pma cfg 2 issue and we are right on this to get it green again, but ci_check should be relatively ok.)

File: docs/VerifPlans/Simulation/privileged_spec/CV32E40SX_PMA_VerifPlan.json
Comment (silabs-krdosvik) [reply docs/VerifPlans/Simulation/privileged_spec/CV32E40SX_PMA_VerifPlan.json:1]: True. Hem, I guess we should have a better system for it, somehow??

**Pull request merged**
